spotify_client_credentials_flow.py

```diff
@@ -0,0 +1,28 @@
+""" Client Credentials Flow
+
+The Client Credentials flow is used in server-to-server authentication. Only endpoints that do not
+access user information can be accessed. The advantage here in comparison with requests to the Web
+API made without an access token, is that a higher rate limit is applied.
+https://spotipy.readthedocs.io/en/2.16.1/#client-credentials-flow"""
+
+from dotenv import load_dotenv
+
+load_dotenv()
+
+import spotipy
+from spotipy.oauth2 import SpotifyClientCredentials
+
+auth_manager = SpotifyClientCredentials()
+sp = spotipy.Spotify(auth_manager=auth_manager)
+
+playlists = sp.user_playlists("spotify")
+while playlists:
```

Comment:
I see that this code is borrowed from https://spotipy.readthedocs.io/en/2.16.1/#api-reference. I see a couple of infelicities about it. This is the less nice of the two.

This idiom of iterating over the spine of a linked-list of playlists is non-pythonic - it looks weird and special purpose. How about transforming into something more familiar:
```python
def all_playlists( sp ):
    """Yields all the playlists in linked order"""
    playlists = sp.user_playlists('spotify')
    while playlists:
        yield playlists
        if playlists['next']:
            playlists = sp.next(playlists)
        else:
            playlists = None
 ```
 
 Then we can rephrase our code as:
 
```python
for playlists in all_playlists( sp ):
    for i, playlist in enumerate(playlists['items']):
        offset = i + 1 + playlists["offset"]
        print( f'''{offset:4} {playlist["uri"]} {playlist["name"]}''' )
```